core/db/queries.py

```diff
@@ -238,13 +258,23 @@ def outcome_data_query(base_query: Query) -> Query:
     return extended_query
 
 
-def outcome_dim_query(base_query: Query) -> Query:
+def outcome_dim_query(base_query: Query, join_outcome_info=False) -> Query:
     """
     Extend base query to select specified columns for OutcomeDim.
 
     :param base_query: SQLAlchemy Query of core tables with filters applied.
+    :param join_outcome_info: boolean of whether to join OutcomeData and OutcomeDim
     :return: updated query.
     """
+    if join_outcome_info:
+        base_query = base_query.join(  # left outer join: Outcomes is child of Project and hence optional
+            ents.OutcomeData,
+            or_(
+                ents.Project.id == ents.OutcomeData.project_id,
```

Comment:
I think this _might_ now need to match the join code in `query_extend_with_outcome_filter `. I think this is actually a mistake in the code that was already here before this ticket, and we just never noticed.

But I'm not 100%, might be worth updating/replacing with the function call and check if everything still works